### PR TITLE
Email and password validation Issue #185

### PR DESCRIPTION
**Pull request by vishwaszadte**
#### ADDED
- Added `validator` as a dependency

#### CHANGED
- Changed `userLogin` controller to validate email
- Changed `userSignup` controller to validate email as well as password

## How to test 
1. In Postman, create a Post request on the following URLs -> `http://localhost:3001/user/signup` (for email, password) and `http://localhost:3001/user/login` (for email) 
2. Try invalid email and/or weak password (not containing 8 chars or without a number or special character) in the request body

### CODE REVIEW

**Review by lugenx — CHANGES_REQUESTED** [review]
thank you, @vishwaszadte, this looks great. I just have one comment about changing the error message. I will wait for @tsimian also review this before the merge.

**Review by tsimian — APPROVED** [review]
Looks good to me @lugenx. Nice work @vishwaszadte!

File: server/src/controller/userController.js [hunk 1]
```diff
@@ -26,6 +34,29 @@ const userLogin = async (req, res) => {
...
+    minNumbers: 1,
+    minSymbols: 1,
+    minLowercase: 0,
+    minUppercase: 0,
+  });
+
+  if (!isValidPassword) {
+    return res.status(400).json({
+      error:
+        "Password not strong enough. A strong password is minimum 8 characters long containing at least 1 number and 1 special character.",
```
[the remark this answers is not in the document]
Comment (vishwaszadte) [reply server/src/controller/userController.js]: @lugenx I've made the requested change

**Comment by vishwaszadte** [comment]
>

**Review by lugenx — APPROVED** [review]
Great work @vishwaszadte, thank you so much 👍

**Pull request merged**